soundcard/utils.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+def match_device(id, devices):
+    """Find id in a list of devices.
+
+    id can be a platfom specific id, a substring of the device name, or a
+    fuzzy-matched pattern for the microphone name.
+    """
+    devices_by_id = {device.id: device for device in devices}
+    real_devices_by_name = {
+        device.name: device for device in devices
+        if not getattr(device, 'isloopback', True)}
```

Comment:
Why are you using `getattr` here? Can there be a case where `isloopback` is not available? (I think that's a bug that would need fixing)

Reply:
Wait, I get it: that's for speakers, right?

Reply:
Right, class _Speaker does not have a property `isloopback`, in all three implementations. And I think it makes sense to leave it that way: What should its meaning be? I prefer to use getattr() instead.